exercises/tom_clayton/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,114 @@
+#include "Shapesorter.h"
+#include <iostream>
+#include <algorithm>
+#include <vector>
+
+//ShapeSorter constructor
+ShapeSorter::ShapeSorter(std::vector<Shape> shpVect) {
+  m_shpVect = shpVect;
+  m_vecLen = shpVect.size();
+}
+
+// Typedef to be used to iterate through the vector
+typedef std::vector<Shape>::size_type VecSz;
+
+/** Print out all shapes of a given type, or an error message if no
+ *  shapes of that type are found
+ *  @param string type for type of string
+ */ 
+const void ShapeSorter::printShapesType(const std::string type) {
+  bool flag = false;
+  for(VecSz i = 0; i < getVecLen(); ++i) {

```

Comment:
You could use a range-based for loop (I don't think they are in the book but I would look into them if you haven't seen them before). For example this would be:
`for(const auto &shape: m_shpVect)`